core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -611,6 +611,8 @@ class KafkaServer(val config: KafkaConfig, time: Time = Time.SYSTEM, threadNameP
         AppInfoParser.unregisterAppInfo(jmxPrefix, config.brokerId.toString)
```

Comment:
An existing issue, but it seems like this line should be executed before `brokerState.newState(NotRunning)` and we should wrap a `CoreUtils.swallow` around it to be safe.